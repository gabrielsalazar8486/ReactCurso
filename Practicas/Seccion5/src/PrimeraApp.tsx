import PropTypes from 'prop-types'

type props = {
  saludo:string,
  subtitulo:string
}

const PrimeraApp = ( {saludo, subtitulo}:props) => {
  return (
    <>
      <h1>{ saludo }</h1>
      <p>{ subtitulo }</p>
    </>
  )
};

PrimeraApp.propTypes = {
  saludo:PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
  subtitulo: 'Subtitulo por defecto'
}

export default PrimeraApp;