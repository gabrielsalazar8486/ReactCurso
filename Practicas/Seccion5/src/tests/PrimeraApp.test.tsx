import PrimeraApp from "../PrimeraApp"
import { shallow } from "enzyme"

describe('PrimeraApp test', ()=>{

  test('debe de mostrar <PrimeraApp /> correctamente', ()=>{
    const saludo = 'Hola soy Monkey D. Luffy'
    const wrapper = shallow(<PrimeraApp saludo={saludo} />)

    expect( wrapper ).toMatchSnapshot();
  })

  test('debe de mostrar el sub titulo enviado por props', ()=>{
    const saludo = 'Hola soy Monkey D. Luffy'
    const subtitulo = 'El rey de los piratas'
    const wrapper = shallow(
      <PrimeraApp
        saludo={saludo}
        subtitulo={subtitulo}
      />
    )

    const textoParrafo = wrapper.find('p').text()
    expect( textoParrafo ).toBe(subtitulo);
  })
})