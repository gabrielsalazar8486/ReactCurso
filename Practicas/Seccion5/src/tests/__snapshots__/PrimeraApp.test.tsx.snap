// Vitest Snapshot v1

exports[`PrimeraApp test > debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Monkey D. Luffy
  </h1>
  <p>
    Subtitulo por defecto
  </p>
</Fragment>
`;

exports[`PrimeraApp test > debe de mostrar el sub titulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola soy Monkey D. Luffy
  </h1>
  <p>
    El rey de los piratas
  </p>
</Fragment>
`;
